src/components/Event/EventContainer.tsx

```diff
@@ -0,0 +1,99 @@
+import { useState } from "react";
+import EventBox, { type EventBoxProps } from "./EventBox";
+
+const Faculties = {
+  1: "Engineer",
+  2: "Science",
+  3: "Arts",
+};
+
+const EventContainer = ({ Events }: { Events: EventBoxProps[] }) => {
+  const [selectedFaculty, setSelectedFaculty] = useState<string | null>(null);
+  const [currEvents, setCurrEvents] = useState<EventBoxProps[]>(Events);
+  const [searchQuery, setSearchQuery] = useState<string>("");
+
+  const handleSearchChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newSearchQuery = e.target.value;
+    setSearchQuery(newSearchQuery);
+
+    const filteredEvents = Events.filter((event) =>
+      event.EventName.toLowerCase().includes(newSearchQuery.toLowerCase())
+    );
+
+    const finalFilteredEvents =
+      selectedFaculty && selectedFaculty !== "คณะ / Faculty"
+        ? filteredEvents.filter(
+            (event) => event.EngFaculty === Faculties[selectedFaculty]
+          )
+        : filteredEvents;
```

Comment:
I can follow the logic implemented here, but the quality of this code needs improvement. 

Considering potential future scenarios, such as new members joining SGCU or revisiting this code six months down the line, it's crucial to ensure its simplicity and clarity. 

let's go by a simple way like this
```suggestion
  let filteredEvents = Events
  if (selectedValue !== "คณะ / Faculty") 
        filteredEvents = Events.filter(
          (event) => event.EngFaculty === Faculties[selectedValue]
        );
```

and imagine, our beloved `อ.1` or `ฝ่ายทะเบียน` asks us to alter 'คณะ / Faculty' to 'เลือกคณะ / Choose your faculty.' 
As a humble software engineer, it's all too easy to make an oversight, like simply changing the wording at [line 82](https://github.com/isd-sgcu/oph66-frontend/pull/23/files#diff-d6a15aba8c64466da98664206bd709eee4ac1da6c8fd6b940d60d985f4e55aa1R82), committing, and pushing the update, only to face backlash from users unable to select their desired faculty.

The issue lies in the usage of 'คณะ / Faculty' within comparison statements at [line 82](https://github.com/isd-sgcu/oph66-frontend/pull/23/files#diff-d6a15aba8c64466da98664206bd709eee4ac1da6c8fd6b940d60d985f4e55aa1R82) and also at [line 39](https://github.com/isd-sgcu/oph66-frontend/pull/23/files#diff-d6a15aba8c64466da98664206bd709eee4ac1da6c8fd6b940d60d985f4e55aa1R39). It's easy for us or our colleagues to overlook these instances.

To fortify this logic, let's enhance it by defining a constant:

```ts
const DEFAULT_CHOSEN_FACULTY_VALUE = "คณะ / Faculty";
```

By replacing all instances of the literal string with this variable, any future alterations in requirements become straightforward. We can simply modify the `DEFAULT_CHOSEN_FACULTY_VALUE` variable, ensuring seamless adaptability to changing needs.